src/Alipay/Provider.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+namespace SocialiteProviders\Alipay;
+
+use GuzzleHttp\RequestOptions;
+use SocialiteProviders\Manager\Contracts\ConfigInterface;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'alipay';
+
+    /**
+     * The scopes being requested.
+     *
+     * @var array
+     */
+    protected $scopes = ['auth_user'];
+
+    protected $baseUrl = 'https://openapi.alipay.com/gateway.do';
+    protected $authUrl = 'https://openauth.alipay.com/oauth2/publicAppAuthorize.htm';
+
+    protected $apiVersion = '1.0';
+    protected $signType = 'RSA2';
+    protected $postCharset = 'UTF-8';
+    protected $format = 'json';
+    protected $sandbox = false;
+
+    /**
+     * @param \SocialiteProviders\Manager\Contracts\OAuth1\ProviderInterface|\SocialiteProviders\Manager\Contracts\OAuth2\ProviderInterface $config
+     */
+    public function setConfig(ConfigInterface $config)
+    {
+        parent::setConfig($config);
+
+        $this->sandbox = $this->config['sandbox'] ?? false;
+        if ($this->sandbox) {
+            $this->baseUrl = 'https://openapi.alipaydev.com/gateway.do';
+            $this->authUrl = 'https://openauth.alipaydev.com/oauth2/publicAppAuthorize.htm';
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    public static function additionalConfigKeys()
+    {
+        return ['sandbox'];
+    }
+
+    /**
+     * {@inheritdoc}.
+     *
+     * @see \Laravel\Socialite\Two\AbstractProvider::getAuthUrl()
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->authUrl, $state);
+    }
+
+    /**
+     * {@inheritdoc}.
+     *
+     * @see \Laravel\Socialite\Two\AbstractProvider::getTokenUrl()
+     */
+    protected function getTokenUrl()
+    {
+        return $this->baseUrl;
+    }
+
+    /**
+     * {@inheritdoc}.
+     *
+     * @see \Laravel\Socialite\Two\AbstractProvider::getUserByToken()
+     */
+    protected function getUserByToken($token)
+    {
+        $params = $this->getPublicFields('alipay.user.info.share');
+        $params += ['auth_token' => $token];
+        $params['sign'] = $this->generateSign($params);
+
+        $response = $this->getHttpClient()->post(
+            $this->baseUrl,
+            [
+                RequestOptions::FORM_PARAMS => $params,
+                RequestOptions::HEADERS     => [
+                    'content-type' => 'application/x-www-form-urlencoded;charset=utf-8',
+                ],
+            ]
+        );
+
+        $response = json_decode($response->getBody()->getContents(), true);
+
+        if (!empty($response['error_response']) || empty($response['alipay_user_info_share_response'])) {
+            throw new \InvalidArgumentException('You have error! '.\json_encode($response, JSON_UNESCAPED_UNICODE));
```

Comment:
```suggestion
            throw new \InvalidArgumentException('Error getting apipay user details '.\json_encode($response, JSON_UNESCAPED_UNICODE));
```

Can we have a more helpful message here please?